bayes/Calibration.py

```diff
@@ -0,0 +1,143 @@
+# ---------
+# AUTHOR: Atul Agrawal (atul.agrawal@tum.de)
+# ---------
+import numpy as np
+import torch as th
+
+import pyro
+import pyro.distributions as dist
+from numpy.core._multiarray_umath import ndarray
+from pyro.infer import EmpiricalMarginal, Importance, NUTS, MCMC
+
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+
+class Inference:
+    def __init__(self, prior_dist, prior_hyperparameters, forward_solve, fw_input, observed_data, obs_noise_dist,
+                 obs_noise_parameters):
+        """
+
+        :param prior_dist: [string] Specify the prior distribution
+        :param prior_hyperparameters: [2x1] Specify the hyperparameters of the prior dist.
+        :param forward_solve:
+            * A wrapper to the forward solve to be provided here which internally calls the forward solve,
+            takes in known input given by fw_input and takes in parameters to be calibrated and returns the model output.
+
+        :param fw_input: [Dict type] ('known_parameters': , 'sensors': ,'time_steps': )
+        :param observed_data [N,] [tensor]: The observed noisy data
+        :param obs_noise_dist: [string] - Specify the noise distribution
+        :param obs_noise_parameters: Hyperaparameters of the noise model
+        """
+        self.prior_dist = prior_dist
+        self.prior_hyperparameters = prior_hyperparameters
+        self.forward_solve = forward_solve
+        self.fw_input = fw_input
+        self.observed_data = observed_data
+        self.obs_noise_dist = obs_noise_dist
+        self.obs_noise_parameters = obs_noise_parameters
+
+    def model(self, observed_data):
+        """
+        Model to construct prior, likelihood and the posterior with the supplied Observed data
```

Comment:
can you have a look at the inferenceproblem.py I think some of the features could be included there